crates/oxc_linter/src/rules/eslint/no_bitwise.rs

```diff
@@ -57,10 +57,7 @@ impl Rule for NoBitwise {
                 .and_then(|v| v.get("allow"))
                 .and_then(serde_json::Value::as_array)
                 .map(|v| {
-                    v.iter()
-                        .filter_map(serde_json::Value::as_str)
-                        .map(ToString::to_string)
-                        .collect()
+                    v.iter().filter_map(serde_json::Value::as_str).map(CompactStr::from).collect()
```

Comment:
non-blocking: can we also sort this list then use a binary search in `allowed_operator`?

Reply:
I think there's a lot of these that could be rewritten, I'll see if I can batch them together

Reply:
@DonIsaac I think this is a bigger discussion, I posted some thoughts in https://github.com/oxc-project/oxc/issues/6408